source/base/exceptions.cc

```diff
@@ -357,18 +357,24 @@ namespace StandardExceptions
     error_name[0] = '\0';
     int resulting_length = MPI_MAX_ERROR_STRING;
 
-    // get the string name of the error code by first converting it to an error
-    // class.
-    int error_class = 0;
-    int ierr = MPI_Error_class (error_code, &error_class);
-
-    // Check the output of the error printing functions. If either MPI function
-    // fails we should just print a less descriptive message.
-    bool error_name_known = ierr == MPI_SUCCESS;
-    if (error_name_known)
+    bool error_name_known = false;
+    // workaround for Open MPI 1.6.5 not accepting
+    // MPI_ERR_LASTCODE in MPI_Error_class
+    if (error_code < MPI_ERR_LASTCODE)
```

Comment:
The standard says that `error codes` have to satisfy `error code<=MPI_ERR_LASTCODE`. 
Since the error message for `MPI_ERR_LASTCODE` is more or less "Unknown error" on all MPI implementations I've seen, I consider handling this error code ourselves a valid option to let OpenMPI 1.6.5 pass as well.